src/server/JasmineGraphServer.cpp

```diff
@@ -935,371 +947,169 @@ bool JasmineGraphServer::batchUploadFile(std::string host, int port, int dataPor
-                    }
-                }
-            }
-            int count = 0;
-
-            while (true) {
-                result_wr = write(sockfd, JasmineGraphInstanceProtocol::FILE_RECV_CHK.c_str(),
-                                  JasmineGraphInstanceProtocol::FILE_RECV_CHK.size());
+    string response = Utils::read_str_trim_wrapper(sockfd, data, FED_DATA_LENGTH);
+    if (response.compare(JasmineGraphInstanceProtocol::HANDSHAKE_OK) != 0) {
+        server_logger.error("Incorrect response. Expected: " + JasmineGraphInstanceProtocol::HANDSHAKE_OK +
+                            " ; Received: " + response);
+        close(sockfd);
+        return false;
+    }
+    server_logger.info("Received: " + response);
 
-                if (result_wr < 0) {
-                    server_logger.log("Error writing to socket", "error");
-                }
+    if (!Utils::send_str_wrapper(sockfd, masterIP)) {
+        close(sockfd);
+        return false;
+    }
+    server_logger.info("Sent: " + masterIP);
 
-                server_logger.log("Sent : " + JasmineGraphInstanceProtocol::FILE_RECV_CHK, "info");
-                server_logger.log("Checking if file is received", "info");
-                bzero(data, FED_DATA_LENGTH + 1);
-                read(sockfd, data, FED_DATA_LENGTH);
-                response = (data);
-                if (response.compare(JasmineGraphInstanceProtocol::FILE_RECV_WAIT) == 0) {
-                    server_logger.log("Received : " + JasmineGraphInstanceProtocol::FILE_RECV_WAIT, "info");
-                    server_logger.log("Checking file status : " + to_string(count), "info");
-                    count++;
-                    sleep(1);
-                    continue;
-                } else if (response.compare(JasmineGraphInstanceProtocol::FILE_ACK) == 0) {
-                    server_logger.log("Received : " + JasmineGraphInstanceProtocol::FILE_ACK, "info");
-                    server_logger.log("File transfer completed for file : " + filePath, "info");
-                    break;
-                }
-            };
-            // Next we wait till the batch upload completes
-            while (true) {
-                result_wr = write(sockfd, JasmineGraphInstanceProtocol::BATCH_UPLOAD_CHK.c_str(),
-                                  JasmineGraphInstanceProtocol::BATCH_UPLOAD_CHK.size());
-
-                if (result_wr < 0) {
-                    server_logger.log("Error writing to socket", "error");
-                }
+    response = Utils::read_str_trim_wrapper(sockfd, data, FED_DATA_LENGTH);
+    if (response.compare(JasmineGraphInstanceProtocol::HOST_OK) != 0) {
+        server_logger.error("Incorrect response. Expected: " + JasmineGraphInstanceProtocol::HOST_OK +
+                            " ; Received: " + response);
+        close(sockfd);
+        return false;
+    }
+    server_logger.info("Received: " + response);
```

Comment:
Repeating code segments for initial handshake
sending -> HANDSHAKE
recieving -> HANDSHAKE_OK
sending -> masterIp
reciving -> HOST_OK

Let's move this to a seperate function or two

Reply:
Will be included in next PR

Reply:
Ok